### PR TITLE
[uss_qualifier] constraint reference sync: check created references can be searched - 3/n

### PR DESCRIPTION
**Pull request by Shastick**
Adds constraint reference search to the synchronization scenario.

### CODE REVIEW

**Comment by Shastick** [comment]
Opening for review (it sits on top of #705)

File: monitoring/monitorlib/schema_validation.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class F3548_21(str, Enum):
     )
     GetConstraintReferenceResponse = "components.schemas.GetConstraintReferenceResponse"
     QueryConstraintReferenceResponse = (
-        "components.schemas.QueryConstraintReferenceResponse"
+        "components.schemas.QueryConstraintReferencesResponse"
```
Comment (mickmis) [inline monitoring/monitorlib/schema_validation.py:78]: Update also the field name.

File: monitoring/uss_qualifier/scenarios/astm/utm/dss/synchronization/constraint_ref_synchronization.py [hunk 2]
```diff
@@ -125,9 +125,9 @@ def run(self, context: ExecutionContext):
         self._create_cr_with_params(self._cr_params)
         self.end_test_step()
 
-        self.begin_test_step("Retrieve newly created CR")
-        self._query_secondaries_and_compare(
-            "Newly created CR can be consistently retrieved from all DSS instances",
```
Comment (mickmis) [inline monitoring/uss_qualifier/scenarios/astm/utm/dss/synchronization/constraint_ref_synchronization.py:130]: This removes the step *Retrieve newly created CR*.

File: monitoring/uss_qualifier/scenarios/astm/utm/dss/fragments/cr/crud/search_correct.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## [Search query succeeds](./search_query.md)
+
+Check query succeeds.
+
+## 🛑 Search constraint reference response format conforms to spec check
+
+The response to a successful constraint reference search query is expected to conform to the format defined by the OpenAPI specification under the `A3.1` Annex of ASTM F3548−21.
+
+If it does not, the DSS is failing to implement **[astm.f3548.v21.DSS0005,3](../../../../../../../requirements/astm/f3548/v21.md)**.
```
[the remark this answers is not in the document]
Comment (Shastick) [reply monitoring/uss_qualifier/scenarios/astm/utm/dss/fragments/cr/crud/search_correct.md:13]: Correct, good catch

File: monitoring/uss_qualifier/scenarios/astm/utm/dss/synchronization/constraint_ref_synchronization.py (hunk 2 above)
Comment (Shastick) [reply monitoring/uss_qualifier/scenarios/astm/utm/dss/synchronization/constraint_ref_synchronization.py:130]: Good catch.

**Review by mickmis — APPROVED** [review]

**Pull request merged**